### PR TITLE
[# 149463909] Add examples to the Factree README

### PR DESCRIPTION
**Pull request by jstrater**
Adds much-needed examples to the Factree README. There are no code changes, just more docs and a patch version bump.

[Tracker #149463909](https://www.pivotaltracker.com/story/show/149463909)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +23,116 @@ Or install it yourself as:
...
+*For details, check out the [API documentation](http://www.rubydoc.info/gems/factree).*
+
+### Finding paths through a decision function
+
+Factree provides tools for making choices based on a set of facts that are not yet known. You write a decision function that takes a set facts and returns a conclusion. Factree will run your function and make sure it has all of the facts it needs to complete. If any are missing, Factree will tell you what's needed to continue.
+
+For example, say I want to pick an animal based on its attributes. First I'll write a function to make the decision.
+
+```ruby
+decide = ->(facts) do
```
Comment (RoryO) [inline README.md]: Does `find_path` use `call`? If so you can add some examples which specify other callable types such as method references, blocks and objects that implement .call like

```ruby
class IsHerbivore
  def self.call
  end
end
```

When you can do with` Factree.find_path mammal: false, IsHerbivore`

Also 
```
Factree.find_path mammal: false do
end
```
should also work

**Review by RoryO — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (jstrater) [reply README.md]: I like the idea. `find_path` takes a block, so it's not exactly the same as passing a `#call`able object directly, but it's pretty close. I'm actually using method references with this in Jarvis.

**Comment by jstrater** [comment]
Hey @RoryO, I added a note on using method references, plus some other information dealing with decision function organization. Mind taking another look?

**Comment by jstrater** [comment]
@RoryO bump

**Review by RoryO — APPROVED** [review]
sorry

**Comment by jstrater** [comment]
@RoryO No problem. For some reason I wasn't the owner on this story, and it dropped off my radar for a while.

**Pull request merged**